src/main/java/org/scijava/log/DefaultLogger.java

```diff
@@ -116,29 +111,19 @@ public void alwaysLog(final int level, final Object msg, final Throwable t) {
 
 	@Override
 	public void addLogListener(final LogListener l) {
-		if (listeners == null) initListeners();
-		synchronized (listeners) {
-			listeners.add(l);
-			cacheListeners();
-		}
+		listeners.add(Objects.requireNonNull(l));
```

Comment:
I agree that we should be requiring non-null here at this level. However, that is a separate conceptual change, which would be better as its own commit. Or else the commit message here should state that it is also changing that behavior.